substrate/primitives/version/src/lib.rs

```diff
@@ -230,9 +221,394 @@ pub struct RuntimeVersion {
+impl serde::Serialize for RuntimeVersion {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		use serde::ser::SerializeStruct;
+
+		let mut state = serializer.serialize_struct("RuntimeVersion", 9)?;
+		state.serialize_field("specName", &self.spec_name)?;
+		state.serialize_field("implName", &self.impl_name)?;
+		state.serialize_field("authoringVersion", &self.authoring_version)?;
+		state.serialize_field("specVersion", &self.spec_version)?;
+		state.serialize_field("implVersion", &self.impl_version)?;
+		state.serialize_field("apis", {
+			struct SerializeWith<'a>(&'a ApisVec);
+
+			impl<'a> serde::Serialize for SerializeWith<'a> {
+				fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+				where
+					S: serde::Serializer,
+				{
+					apis_serialize::serialize(self.0, serializer)
+				}
+			}
+
+			&SerializeWith(&self.apis)
+		})?;
+		state.serialize_field("transactionVersion", &self.transaction_version)?;
+		state.serialize_field("systemVersion", &self.system_version)?;
+		state.serialize_field("stateVersion", &self.system_version)?;
+		state.end()
+	}
+}
+
+// Manual implementation in order to allow both old `stateVersion` and new `systemVersion` to be
+// present at the same time
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for RuntimeVersion {
+	fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+	where
+		D: serde::Deserializer<'de>,
+	{
+		use core::marker::PhantomData;
+
+		enum Field {
+			SpecName,
+			ImplName,
+			AuthoringVersion,
+			SpecVersion,
+			ImplVersion,
+			Apis,
+			TransactionVersion,
+			SystemVersion,
+			Ignore,
+		}
+
+		struct FieldVisitor;
+
+		impl<'de> serde::de::Visitor<'de> for FieldVisitor {
+			type Value = Field;
+
+			fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+				formatter.write_str("field identifier")
+			}
+
+			fn visit_u64<E>(self, value: u64) -> Result<Self::Value, E>
+			where
+				E: serde::de::Error,
+			{
+				match value {
+					0 => Ok(Field::SpecName),
+					1 => Ok(Field::ImplName),
+					2 => Ok(Field::AuthoringVersion),
+					3 => Ok(Field::SpecVersion),
+					4 => Ok(Field::ImplVersion),
+					5 => Ok(Field::Apis),
+					6 => Ok(Field::TransactionVersion),
+					7 => Ok(Field::SystemVersion),
+					_ => Ok(Field::Ignore),
+				}
+			}
+
+			fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+			where
+				E: serde::de::Error,
+			{
+				match value {
+					"specName" => Ok(Field::SpecName),
+					"implName" => Ok(Field::ImplName),
+					"authoringVersion" => Ok(Field::AuthoringVersion),
+					"specVersion" => Ok(Field::SpecVersion),
+					"implVersion" => Ok(Field::ImplVersion),
+					"apis" => Ok(Field::Apis),
+					"transactionVersion" => Ok(Field::TransactionVersion),
+					"systemVersion" | "stateVersion" => Ok(Field::SystemVersion),
+					_ => Ok(Field::Ignore),
+				}
+			}
+
+			fn visit_bytes<E>(self, value: &[u8]) -> Result<Self::Value, E>
```

Comment:
And this?

Reply:
Same here, this is from `Deserialize` derive